ufmt/core.py

```diff
@@ -29,27 +36,57 @@ class Result:
     diff: Optional[str] = None
 
 
-def ufmt_string(path: Path, content: str, config: Config) -> str:
-    content = usort_string(content, config, path)
-    mode = Mode()  # TODO
+def ufmt_string(
+    path: Path,
+    content: str,
+    usort_config: UsortConfig,
+    black_config: Optional[Mode] = None,
+) -> str:
+    content = usort_string(content, usort_config, path)
 
     try:
-        content = format_file_contents(content, fast=False, mode=mode)
+        content = format_file_contents(content, fast=False, mode=black_config or Mode())
     except NothingChanged:
         pass
 
     return content
 
 
+def _make_black_config(path: Path) -> Mode:
+    config_file = find_pyproject_toml((str(path),))
+    if not config_file:
+        return Mode()
```

Comment:
Coverage tests will fail, since the tests currently don't set a `pyproject.toml` and this early exit prevents most of the changes from executing.

Reply:
Yeah, I'd like to see a test or two covering this function. A simple mock of `find_pyproject_toml()` and/or `parse_pyproject_toml()` would suffice, rather than needing to write a `pyproject.toml` to disk in a temp dir.